test/es6ToJSONTest.js

```diff
@@ -113,32 +111,67 @@ describe('#ToJSON', () => {
     const RecursiveEntry = importResult('shr/simple/RecursiveEntry');
     it('should serialize a JSON instance', () => {
       // This one is special cased because you're working with recursive entries
-      const json = getJSON('RecursiveEntry');
+      const json = context.getJSON('RecursiveEntry');
       const entry = RecursiveEntry.fromJSON(json);
       expect(entry).instanceOf(RecursiveEntry);
       let gen_json = entry.toJSON();
-      validateJSON('RecursiveEntry', gen_json);
+      context.validateJSON('RecursiveEntry', gen_json);
       expect(gen_json).to.eql(json);
       
       // Recursive child 1
       const child1 = entry.recursiveEntry[0];
       expect(child1).instanceOf(RecursiveEntry);
       let child1_json = child1.toJSON();
-      validateJSON('RecursiveEntry', child1_json);
+      context.validateJSON('RecursiveEntry', child1_json);
       expect(gen_json).to.eql(json);
 
       // Recursive grandchild 1
       const grandchild1 = child1.recursiveEntry[0];
       expect(grandchild1).instanceOf(RecursiveEntry);
       let grandchild1_json = grandchild1.toJSON();
-      validateJSON('RecursiveEntry', grandchild1_json);
+      context.validateJSON('RecursiveEntry', grandchild1_json);
       expect(gen_json).to.eql(json);
 
       // Recursive child 2
       const child2 = entry.recursiveEntry[1];
       expect(child2).instanceOf(RecursiveEntry);
       let child2_json = child2.toJSON();
-      validateJSON('RecursiveEntry', child2_json);
+      context.validateJSON('RecursiveEntry', child2_json);
+      expect(gen_json).to.eql(json);
+
+      // Recursive child 2 with a null recursive entry
+      child2.recursiveEntry = null;
+      child2_json = child2.toJSON();
+      context.validateJSON('RecursiveEntry', child2_json);
+      gen_json = entry.toJSON();
+      expect(gen_json).to.eql(json);
```

Comment:
This time `gen_json` was re-assigned, but just to `entry.toJSON()` which is how it was originally assigned in line 117.  So... the test still seems redundant.

Reply:
That assertion is testing the `child2.recursiveEntry = null;` assignment on line 140.